doc/neps/nep-template.rst

```diff
@@ -13,15 +13,20 @@ Abstract
 
 The abstract should be a short description of what the NEP will achieve.
 
-
-Detailed description
---------------------
+Motivation
+----------
 
 This section describes the need for the NEP.  It should describe the existing
-problem that it is trying to solve and why this NEP makes the situation better.
-It should include examples of how the new functionality would be used and
-perhaps some use cases.
+problem, who it affects, what it is trying to solve, and why.
+It should include examples of how the new functionality would be used, intended
+use-cases and psuedo-code describing its use.
 
+Literature Review
```

Comment:
I think this is a good section to have, most NEPs do have relevant prior art somewhere. The title isn't optimal though, since it's mostly not literature (that gives the wrong signal to an academic) - how about "Related work"?